Assets/Amplitude/Amplitude.cs

```diff
@@ -9,157 +9,183 @@
 	[DllImport ("__Internal")]
-	private static extern void _Amplitude_appendUserPropertyString(string property, string value);
+	private static extern void _Amplitude_appendUserPropertyString(string instanceName, string property, string value);
 	[DllImport ("__Internal")]
-	private static extern void _Amplitude_appendUserPropertyDict(string property, string values);
+	private static extern void _Amplitude_appendUserPropertyDict(string instanceName, string property, string values);
 	[DllImport ("__Internal")]
-	private static extern void _Amplitude_appendUserPropertyList(string property, string values);
+	private static extern void _Amplitude_appendUserPropertyList(string instanceName, string property, string values);
 	[DllImport ("__Internal")]
-	private static extern void _Amplitude_appendUserPropertyBoolArray(string property, bool[] value, int length);
+	private static extern void _Amplitude_appendUserPropertyBoolArray(string instanceName, string property, bool[] value, int length);
 	[DllImport ("__Internal")]
-	private static extern void _Amplitude_appendUserPropertyDoubleArray(string property, double[] value, int length);
+	private static extern void _Amplitude_appendUserPropertyDoubleArray(string instanceName, string property, double[] value, int length);
 	[DllImport ("__Internal")]
-	private static extern void _Amplitude_appendUserPropertyFloatArray(string property, float[] value, int length);
+	private static extern void _Amplitude_appendUserPropertyFloatArray(string instanceName, string property, float[] value, int length);
 	[DllImport ("__Internal")]
-	private static extern void _Amplitude_appendUserPropertyIntArray(string property, int[] value, int length);
+	private static extern void _Amplitude_appendUserPropertyIntArray(string instanceName, string property, int[] value, int length);
 	[DllImport ("__Internal")]
-	private static extern void _Amplitude_appendUserPropertyLongArray(string property, long[] value, int length);
+	private static extern void _Amplitude_appendUserPropertyLongArray(string instanceName, string property, long[] value, int length);
 	[DllImport ("__Internal")]
-	private static extern void _Amplitude_appendUserPropertyStringArray(string property, string[] value, int length);
+	private static extern void _Amplitude_appendUserPropertyStringArray(string instanceName, string property, string[] value, int length);
 #endif
 
+	public static Amplitude getInstance() {
+		return getInstance(null);
+	}
+	public static Amplitude getInstance(string instanceName) {
+		string instanceKey = instanceName;
+		if (string.IsNullOrEmpty(instanceKey)) {
+			instanceKey = "$default_instance";
+		}
+
+		if (instances == null) {
+			instances = new Dictionary<string, Amplitude>();
+		}
+
+		Amplitude instance;
+		if (instances.TryGetValue(instanceKey, out instance)) {
+			return instance;
+		}
+		else {
+			instance = new Amplitude(instanceName);
```

Comment:
This is where I leave `instanceName` as `null` if it's the default instance, but use `$default_instance` as the key in the dictionary

Reply:
are we worried about concurrency at all?